### PR TITLE
Avoid sending internal server error message on state validation failure

### PR DESCRIPTION
**Pull request by madkinsz**
We should not be including the exception in the response to the user as it could share sensitive details from the server.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Review by anticorrelator — COMMENTED** [review]
This looks reasonable to me

File: src/prefect/orion/orchestration/rules.py [hunk 1]
```diff
@@ -237,11 +238,12 @@ async def validate_proposed_state(
             await self._validate_proposed_state()
             return
         except Exception as exc:
-            logger.exception("Encountered error during state validation")
+            error_key = hashlib.md5(str(exc).encode())
+            error_msg = f"Internal server error while validating state ({error_key}"
```
Comment (peytonrunyan) [inline src/prefect/orion/orchestration/rules.py:242]: Possibly a dumb question, but what are we doing with this error key?

Comment (madkinsz) [reply src/prefect/orion/orchestration/rules.py:242]: The user could provide the key to a server admin to lookup the internal error.

**Comment by madkinsz** [comment]
This is more critical in Cloud than the open source.

**Pull request closed without merging**